revolv/project/tests/test_integration.py

```diff
@@ -0,0 +1,193 @@
+import mock
+import json
+from django.test import TestCase
+from django_webtest import WebTest
+from revolv.lib.testing import TestUserMixin, UserTestingMixin
+from revolv.project.models import Project, ProjectUpdate
+
+
+class PostProjectUpdatesTest(TestUserMixin, UserTestingMixin, TestCase):
+
+    def assert_user_can_create_project_update(self):
+        """
+        Tests that administrators and ambassadors can create project updates. This assert function logs in, creates a project,
+        posts an update to the project, and then verifies that the posted update exists.
+        """
+        response = self.send_test_user_login_request()
+        self.assertUserAuthed(response)
+        project = Project.factories.base.create()
+        response = self.client.post('/project/%d/update' % project.pk, {'update_text': 'This is an update'})
+        new_updates = ProjectUpdate.objects.filter(update_text = 'This is an update')
+        self.assertEqual(len(new_updates),1)
+        self.assertEqual(new_updates[0].project, project)
+
+    def test_admin_form_submission(self):
+        """
+        Tests that administrators can create project updates.
+        This function makes an administrator and then calls the assert_user_can_create_project_update function.
+        """
+        self.test_profile.make_administrator()
+        self.assert_user_can_create_project_update()
+
+    def test_ambassador_form_submission(self):
+        """
+        Tests that ambassadors can create project updates.
+        This function makes an ambassador and then calls the assert_user_can_create_project_update function.
+        """
+        self.test_profile.make_ambassador()
+        self.assert_user_can_create_project_update()
+
+    def test_donor_form_submission(self):
+        """
+        Tests that donors can't create project updates.
+        This function makes a donor, logs in, creates a project, and then attempts to post an update to the project.
+        """
+        self.test_profile.make_donor()
+        response = self.send_test_user_login_request()
+        self.assertUserAuthed(response)
+        project = Project.factories.base.create()
+        response = self.client.post('/project/%d/update' % project.pk, {'update_text': 'This is an update'})
+        
+        # the deny_access method is called in the view from the userdatamixin, which returns a redirect to the homepage
+        self.assertEqual(response.status_code, 302)
+        
+class EditProjectUpdatesTest(TestUserMixin, UserTestingMixin, TestCase):
+
+    def assert_make_update(self, project, text):
```

Comment:
Sorry to be pedantic about this, but the point of this function isn't really to assert anything. Sure, it asserts that the user is authed at one point, but the point is to make a new update, so I think it's fine to call it `make_update`. I think that methods whose point is to check that something is the same as expected (or even check that more than one thing is the same as expected) should start with `assert_` but methods like this don't need to.